packages/headless/src/app/commerce-engine/commerce-engine.ts

```diff
@@ -113,24 +113,30 @@ export function buildCommerceEngine(
     reducers: commerceEngineReducers,
   };
 
-  const engine = buildEngine(augmentedOptions, thunkArguments);
+  const internalEngine = buildEngine(augmentedOptions, thunkArguments);
+  const {state: _, ...engine} = internalEngine;
 
   engine.dispatch(setContext(options.configuration.context));
 
   return {
     ...engine,
 
-    get state() {
-      return engine.state;
+    get [stateKey]() {
```

Comment:
TIL you can use [var] with a getter like so. I thought you had to use Object.defineProperty to define a prop with  Symbol as key
But this is much betta